flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/CreditBasedPartitionRequestClientHandlerTest.java

```diff
@@ -443,10 +587,13 @@ private static BufferResponse createBufferResponse(
 		// Skip general header bytes
 		serialized.readBytes(NettyMessage.FRAME_HEADER_LENGTH);
 
-		// Deserialize the bytes again. We have to go this way, because we only partly deserialize
-		// the header of the response and wait for a buffer from the buffer pool to copy the payload
-		// data into.
-		BufferResponse deserialized = BufferResponse.readFrom(serialized);
+		// Deserialize the bytes again. We have to go this way to ensure the data buffer part
+		// is consistent with the input channel sent to.
```

Comment:
I think the previous comment makes more sense. `because we only partly deserialize the deader of the response and wait for a buffer from the allocator to copy the payload data into.

Reply:
I think the previous comments is trying to explain why it needs to deserialize the bytes again: since it need to reach the state that the message is deserialized but not copied to a Flink Buffer yet. I thought that the comment might not be consistent with current implementation.